src/bias/MetaD.cpp

```diff
@@ -1998,27 +2190,83 @@ double MetaD::getTransitionBarrierBias() {
   }
 }
 
-void MetaD::updateFrequencyAdaptiveStride() {
+void MetaD::updateFrequencyAdaptiveStride()
+{
   plumed_massert(freq_adaptive_,"should only be used if frequency adaptive metadynamics is enabled");
-  plumed_massert(acceleration,"frequency adaptive metadynamics can only be used if the acceleration factor is calculated");
-  const double mean_acc = acc/((double) getStep());
+  plumed_massert(acceleration_,"frequency adaptive metadynamics can only be used if the acceleration factor is calculated");
+  const double mean_acc = acc_/((double) getStep());
   int tmp_stride= stride_*floor((mean_acc/fa_min_acceleration_)+0.5);
   if(mean_acc >= fa_min_acceleration_) {
-    if(tmp_stride > current_stride) {current_stride = tmp_stride;}
+    if(tmp_stride > current_stride_) {current_stride_ = tmp_stride;}
   }
-  if(fa_max_stride_!=0 && current_stride>fa_max_stride_) {
-    current_stride=fa_max_stride_;
+  if(fa_max_stride_!=0 && current_stride_>fa_max_stride_) {
+    current_stride_=fa_max_stride_;
   }
-  getPntrToComponent("pace")->set(current_stride);
+  getPntrToComponent("pace")->set(current_stride_);
 }
 
 bool MetaD::checkNeedsGradients()const
 {
   if(adaptive_==FlexibleBin::geometry) {
-    if(getStep()%stride_==0 && !isFirstStep) return true;
+    if(getStep()%stride_==0 && !isFirstStep_) return true;
     else return false;
   } else return false;
 }
 
+void MetaD::updateNlist()
+{
+  // no need to check for neighbors
+  if(hills_.size()==0) return;
+
+  // here we generate the neighbor list
+  nlist_hills_.clear();
+  std::vector<Gaussian> local_flat_nl;
+  unsigned nt=OpenMP::getNumThreads();
+
+  #pragma omp parallel num_threads(nt)
+  {
+    std::vector<Gaussian> private_flat_nl;
+    if(hills_.size()<2*nt) nt=1;
```

Comment:
@carlocamilloni should this line be moved before the `#pragma omp parallel`

Reply:
done